core/src/main/java/org/carbondata/core/carbon/datastore/chunk/impl/ColumnGroupDimensionDataChunk.java

```diff
@@ -67,10 +67,29 @@ public ColumnGroupDimensionDataChunk(byte[] dataChunk, DimensionChunkAttributes
   }
 
   /**
-   * Below method to get the data based in row id
+   * Converts to column dictionary integer value
+   * @param rowId
+   * @param columnIndex
+   * @param row
+   * @param info  @return
+   */
+  @Override public int fillConvertedChunkData(int rowId, int columnIndex, int[] row,
+      KeyStructureInfo info) {
+    int start = rowId * chunkAttributes.getColumnValueSize();
+    int sizeInBytes = info.getKeyGenerator().getKeySizeInBytes();
+    byte[] key = new byte[sizeInBytes];
+    System.arraycopy(dataChunk, start, key, 0, sizeInBytes);
+    long[] keyArray = info.getKeyGenerator().getKeyArray(key);
```

Comment:
we can avoid this copy also by changing getKeyArray interface to take start and end position, or wrapping it with ByteBuffer